test/integrationTests/inlayHints.integration.test.ts

```diff
@@ -0,0 +1,76 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+
+import { should, assert } from 'chai';
+import { activateCSharpExtension, isRazorWorkspace, isSlnWithGenerator, restartOmniSharpServer } from './integrationHelpers';
+import testAssetWorkspace from './testAssets/testAssetWorkspace';
+import * as path from 'path';
+
+const chai = require('chai');
+chai.use(require('chai-arrays'));
+chai.use(require('chai-fs'));
+
+suite(`Inlay Hints ${testAssetWorkspace.description}`, function () {
+    let fileUri: vscode.Uri;
+
+    suiteSetup(async function () {
+        should();
+
+        if (isRazorWorkspace(vscode.workspace) || isSlnWithGenerator(vscode.workspace)) {
+            this.skip();
+        }
+
+        const editorConfig = vscode.workspace.getConfiguration('editor');
+        editorConfig.update('inlayHints.enabled', true);
+
+        const csharpConfig = vscode.workspace.getConfiguration('csharp');
+        csharpConfig.update('inlayHints.parameters.enabled', true);
+        csharpConfig.update('inlayHints.parameters.forLiteralParameters', true);
+        csharpConfig.update('inlayHints.parameters.forObjectCreationParameters', true);
+        csharpConfig.update('inlayHints.parameters.forIndexerParameters', true);
+        csharpConfig.update('inlayHints.parameters.forOtherParameters', true);
+        csharpConfig.update('inlayHints.parameters.suppressForParametersThatDifferOnlyBySuffix', true);
+        csharpConfig.update('inlayHints.parameters.suppressForParametersThatMatchMethodIntent', true);
+        csharpConfig.update('inlayHints.parameters.suppressForParametersThatMatchArgumentName', true);
+        csharpConfig.update('inlayHints.types.enabled', true);
+        csharpConfig.update('inlayHints.types.forImplicitVariableTypes', true);
+        csharpConfig.update('inlayHints.types.forLambdaParameterTypes', true);
+        csharpConfig.update('inlayHints.types.forImplicitObjectCreation', true);
+
+        await restartOmniSharpServer();
+        const activation = await activateCSharpExtension();
+        await testAssetWorkspace.restoreAndWait(activation);
+
+        const fileName = 'inlayHints.cs';
+        const projectDirectory = testAssetWorkspace.projects[0].projectDirectoryPath;
+        const filePath = path.join(projectDirectory, fileName);
+        fileUri = vscode.Uri.file(filePath);
+    });
+
+    suiteTeardown(async () => {
+        await testAssetWorkspace.cleanupWorkspace();
+    });
+
+    test("Hints retrieved for region", async () => {
+        await vscode.commands.executeCommand("vscode.open", fileUri);
```

Comment:
If you take the above suggestion, there is no need to open the file in the test.
```suggestion
```